aea/package_manager/base.py

```diff
@@ -0,0 +1,426 @@
+                    self.update_package(package_id=package_id)
+
+        if update_hashes:
+            self._logger.info("Updating hashes")
+            self.dump()
+
+        if sync_needed:
+            self._logger.info("Sync complete")
+        else:
+            self._logger.info("No package was updated.")
+
+        return self
+
+    def add_package(self, package_id: PackageId) -> "PackageManager":
+        """Add packages."""
+
+        author_repo = self.path / package_id.author
+        if not author_repo.exists():
+            author_repo.mkdir()
+            (author_repo / "__init__.py").touch()
+
+        package_type_collection = author_repo / package_id.package_type.to_plural()
+        if not package_type_collection.exists():
+            package_type_collection.mkdir()
+            (package_type_collection / "__init__.py").touch()
+
+        download_path = package_type_collection / package_id.name
+        fetch_ipfs(
+            str(package_id.package_type), package_id.public_id, dest=str(download_path)
+        )
+
+        return self
+
+    def update_package(
+        self,
+        package_id: PackageId,
+    ) -> "PackageManager":
+        """Update package."""
+
+        package_path = self.package_path_from_package_id(package_id=package_id)
+        try:
+            shutil.rmtree(str(package_path))
+        except (OSError, PermissionError) as e:
+            raise PackageUpdateError(f"Cannot update {package_id}") from e
+
+        self.add_package(package_id)
+        return self
+
+    def get_available_package_hashes(
+        self,
+    ) -> OrderedDictType[PackageId, str]:
+        """Returns a mapping object between available packages and their hashes"""
+
+        _packages = OrderedDict()
+        available_packages = DependencyTree.generate(self.path)
+        for _level in available_packages:
+            for package in _level:
+                path = self.package_path_from_package_id(package)
+                package_hash = IPFSHashOnly.get(str(path))
+                _packages[package] = package_hash
+
+        return _packages
+
+    def update_package_hashes(self) -> "PackageManager":
+        """Initialize package.json file."""
+        for package_id, package_hash in self.get_available_package_hashes().items():
+            is_dev_package = package_id in self._dev_packages
+            is_third_party_package = package_id in self._third_party_packages
+            if not is_dev_package and not is_third_party_package:
+                raise PackageNotValid(
+                    f"Found a package which is not listed in the `packages.json` with package id {package_id}"
+                )
+
+            if is_dev_package:
+                if self._dev_packages[package_id] == package_hash:
+                    continue
+
+                self._logger.info(f"Updating hash for {package_id}")
+                self._dev_packages[package_id] = package_hash
```

Comment:
We only want to update the hashes for dev packages

Reply:
Aha yes it is clear to me now